plaso/parsers/google_logging.py

```diff
@@ -0,0 +1,264 @@
+"""Parser for Google-formatted log files."""
+
+from __future__ import unicode_literals
+
+import re
+
+from dfdatetime import time_elements as dfdatetime_time_elements
+
+import pyparsing
+
+from plaso.containers import events
+from plaso.containers import time_events
+from plaso.lib import errors
+from plaso.lib import definitions
+from plaso.parsers import manager
+from plaso.parsers import text_parser
+
+
+
+class GoogleLogEventData(events.EventData):
+  """Convenience class for events from a Google log file.
+
+  See: https://github.com/google/glog
+
+  Attributes:
+    priority (str): the priority of the message - I, W, E or F.
+    thread_id (int): the ID of the thread that recorded the message.
+    file_name (str): the name of the source file that logged the message.
+    line_number (int): the line number in the source file where the logging
+      statement is.
+    message (str): the log message.
+  """
+
+  DATA_TYPE = 'googlelog:log'
+
+  def __init__(self, data_type=DATA_TYPE):
+    """Initializes the event object.
+
+    Args:
+      data_type (Optional[str]): event data type indicator.
+    """
+    super(GoogleLogEventData, self).__init__(data_type=data_type)
+    self.priority = None
+    self.thread_id = None
+    self.file_name = None
+    self.line_number = None
+    self.message = None
+
+
+class GoogleLogParser(text_parser.PyparsingMultiLineTextParser):
+  """A parser for handling google log formatted files.
+
+  See https://google.io/docs/python/guides/logging
+
+  """
+  NAME = 'googlelog'
+
+  DESCRIPTION = 'Parser for handling Google log formatted files.'
+
+  # PyParsing Components used to construct grammars for parsing lines.
+  _PYPARSING_COMPONENTS = {
+      'priority':
+          (pyparsing.oneOf(['I', 'W', 'E', 'F']).setResultsName('priority')),
+      'year': text_parser.PyparsingConstants.FOUR_DIGITS.setResultsName(
+          'year'),
+      'month_number': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'month_number'),
+      'day': text_parser.PyparsingConstants.ONE_OR_TWO_DIGITS.setResultsName(
+          'day'),
+      'hour': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'hour'),
+      'minute': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'minute'),
+      'second': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'second'),
+      'microsecond': pyparsing.Word(pyparsing.nums, exact=6).setParseAction(
+          text_parser.PyParseIntCast).setResultsName('microsecond'),
+      'thread_id': pyparsing.Word(pyparsing.nums).setResultsName('thread_id'),
```

Comment:
id => identifier 

Also see: https://github.com/log2timeline/l2tdocs/blob/master/process/Style-guide.md#naming

Reply:
The parameter in the library is consistently referred to as an 'id' and never as an 'identifier': https://github.com/google/glog/blob/5b4fb63d277795eea3400e3e6af542f3b765f2d2/src/glog/logging.h.in#L293 etc.

I think it's better to match the source here, so keeping this as 'thread_id'

Reply:
Why is this "better"? Why is this unique for this specific format? Why is it not clear from thread_identifier that this field is referenced?

Please add your rationale why to deviate from previous established guidelines.